### ISSUE TITLE
Date component buggy behavior

### ISSUE THREAD

**Issue by chevdor** [hay-kot/homebox#348]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.

### Homebox Version

master

### What is the issue you are experiencing?

Filling up a date (any field) on MacOS/Brave, I tried typing: `12.12.2012` and as soon as I type the `2` of `2012`, the component wipes its content. I tried alternate options such as `12.12.12` without much success.

I added a recording below showing the behavior.
![out](https://user-images.githubusercontent.com/738724/224952976-858b36db-b7d0-4ca1-9237-c305ed370b90.gif)

### How can the maintainer reproduce the issue?

I would try typing a date in a date field. The issue does not seem to be an edge case on a specific date.
I guess the issue is either known or platform specific.
I have tried in the homebox theme as well as the Dracula one.

### Deployment

Docker (Linux)

### Deployment Details

I mentioned Docker/Linux as the closest option but I am using Kubernetes.
I doubt the issue is related to deployement though.

**Comment by chevdor** [comment]
Yes I agree. It is hard to be 100% sure but I tested using a local docker container before doing my k8s setup and __I think__™ it was working fine then (same OS, same browser).

What I know for sure is that I have the issue atm 100% of the time.

My workaround atm is not great, I fill up the year first, then I go back to the month, and finally the day.
It does not work 100% of the time but a retry then usually passes.

**Comment by chevdor** [comment]
So I spinned up a brand new container again and I can see the issue in there as well.

- docker run --rm -it -p 3100:7745 ghcr.io/hay-kot/homebox
- open http://localhost:3100/
- register new account
- create new item
- edit item
- set a date

To ensure there was no issue with local settings that could have been saved (I tested many themes...), also tested again using a local port 3101. Same problem.

**Comment by chevdor** [comment]
I tried with Safari and while the component looks different and fancier, it shows the issue as well.

I tested the demo site and I do **NOT** get the issue there.
I checked the container logs to see a version or a git ref but the logs reports none of those.

My Brave version is:
```
Brave is up to date
[Version 1.49.120 Chromium: 111.0.5563.64 (Official Build) (x86_64)](https://brave.com/latest/)
```

I am using a Docker image that is 3w old. I will try building a fresher one.
```
ghcr.io/hay-kot/homebox           latest        b973f70d7d21   3 weeks ago     38.8MB
```

**Comment by chevdor** [comment]
It was a bit hidden in the UI but found the version displayed under `Profile`:
Version: v0.8.2 ~ Build: 88f9ff90d45a0c59425802ed83106dd909bbb018

**Comment by chevdor** [comment]
I suspect it may have been fixed in [here](https://github.com/hay-kot/homebox/commit/cf536393f5f211a86f61789595ade35e5cc7b6be). I can confirm in a few minutes.

**Comment by chevdor** [comment]
Closing this issue now as I see it is fixed indeed. I tested on `1f47d96`.

**Comment by chevdor** [comment]
In case that helps anyone with this issue before the next version is released, I pushed an image: `chevdor/homebox:20230307-1f47d96`. This is obviously not an official version and it will show as:
<img width="251" alt="image" src="https://user-images.githubusercontent.com/738724/225028731-4156b309-613b-425b-971f-71e3d52acc91.png">

**Comment by hay-kot** [comment]
@chevdor FYI, we publish a nightly tag every time a change is merged into main

`docker pull ghcr.io/hay-kot/homebox:nightly`
